ERROR - 2025-08-25 15:43:35 --> mysqli_sql_exception: Duplicate entry '0' for key 'PRIMARY' in C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\MySQLi\Connection.php:327
Stack trace:
#0 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\MySQLi\Connection.php(327): mysqli->query('INSERT INTO `au...', 0)
#1 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `au...')
#2 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `au...')
#3 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `au...', Array, false)
#4 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
#5 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\BaseModel.php(838): CodeIgniter\Model->doInsert(Array)
#6 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Model.php(797): CodeIgniter\BaseModel->insert(Array, true)
#7 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\app\Controllers\Autores.php(28): CodeIgniter\Model->insert(Array)
#8 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
#9 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
#10 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Boot.php(363): CodeIgniter\CodeIgniter->run()
#12 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
#13 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\public\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
#14 {main}
CRITICAL - 2025-08-25 15:43:35 --> CodeIgniter\Database\Exceptions\DatabaseException: Duplicate entry '0' for key 'PRIMARY'
[Method: POST, Route: agregarAutores]
in SYSTEMPATH\Database\BaseConnection.php on line 684.
 1 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 2 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 3 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 4 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 5 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 6 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
 7 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
 8 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
11 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 15:43:35 --> [Caused by] CodeIgniter\Database\Exceptions\DatabaseException: Duplicate entry '0' for key 'PRIMARY'
in SYSTEMPATH\Database\MySQLi\Connection.php on line 332.
 1 SYSTEMPATH\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'0\', \'Gonzalez\', \'aaron\', \'salvador\')')
 2 SYSTEMPATH\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'0\', \'Gonzalez\', \'aaron\', \'salvador\')')
 3 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 4 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 5 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 6 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 7 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 8 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
 9 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
10 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
11 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
12 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
13 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 15:43:35 --> [Caused by] mysqli_sql_exception: Duplicate entry '0' for key 'PRIMARY'
in SYSTEMPATH\Database\MySQLi\Connection.php on line 327.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(327): mysqli->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'0\', \'Gonzalez\', \'aaron\', \'salvador\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'0\', \'Gonzalez\', \'aaron\', \'salvador\')')
 3 SYSTEMPATH\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'0\', \'Gonzalez\', \'aaron\', \'salvador\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 5 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 6 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 7 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 8 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 9 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
10 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
11 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
13 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
14 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
ERROR - 2025-08-25 15:43:49 --> mysqli_sql_exception: Duplicate entry '0' for key 'PRIMARY' in C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\MySQLi\Connection.php:327
Stack trace:
#0 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\MySQLi\Connection.php(327): mysqli->query('INSERT INTO `au...', 0)
#1 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `au...')
#2 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `au...')
#3 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `au...', Array, false)
#4 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
#5 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\BaseModel.php(838): CodeIgniter\Model->doInsert(Array)
#6 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Model.php(797): CodeIgniter\BaseModel->insert(Array, true)
#7 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\app\Controllers\Autores.php(28): CodeIgniter\Model->insert(Array)
#8 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
#9 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
#10 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Boot.php(363): CodeIgniter\CodeIgniter->run()
#12 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
#13 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\public\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
#14 {main}
CRITICAL - 2025-08-25 15:43:49 --> CodeIgniter\Database\Exceptions\DatabaseException: Duplicate entry '0' for key 'PRIMARY'
[Method: POST, Route: agregarAutores]
in SYSTEMPATH\Database\BaseConnection.php on line 684.
 1 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 2 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 3 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 4 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 5 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 6 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
 7 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
 8 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
11 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 15:43:49 --> [Caused by] CodeIgniter\Database\Exceptions\DatabaseException: Duplicate entry '0' for key 'PRIMARY'
in SYSTEMPATH\Database\MySQLi\Connection.php on line 332.
 1 SYSTEMPATH\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'0\', \'vale\', \'aa\', \'salvador\')')
 2 SYSTEMPATH\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'0\', \'vale\', \'aa\', \'salvador\')')
 3 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 4 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 5 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 6 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 7 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 8 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
 9 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
10 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
11 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
12 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
13 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 15:43:49 --> [Caused by] mysqli_sql_exception: Duplicate entry '0' for key 'PRIMARY'
in SYSTEMPATH\Database\MySQLi\Connection.php on line 327.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(327): mysqli->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'0\', \'vale\', \'aa\', \'salvador\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'0\', \'vale\', \'aa\', \'salvador\')')
 3 SYSTEMPATH\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'0\', \'vale\', \'aa\', \'salvador\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 5 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 6 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 7 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 8 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 9 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
10 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
11 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
13 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
14 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
ERROR - 2025-08-25 15:43:55 --> mysqli_sql_exception: Duplicate entry '0' for key 'PRIMARY' in C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\MySQLi\Connection.php:327
Stack trace:
#0 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\MySQLi\Connection.php(327): mysqli->query('INSERT INTO `au...', 0)
#1 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `au...')
#2 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `au...')
#3 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `au...', Array, false)
#4 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
#5 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\BaseModel.php(838): CodeIgniter\Model->doInsert(Array)
#6 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Model.php(797): CodeIgniter\BaseModel->insert(Array, true)
#7 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\app\Controllers\Autores.php(28): CodeIgniter\Model->insert(Array)
#8 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
#9 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
#10 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Boot.php(363): CodeIgniter\CodeIgniter->run()
#12 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
#13 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\public\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
#14 {main}
CRITICAL - 2025-08-25 15:43:55 --> CodeIgniter\Database\Exceptions\DatabaseException: Duplicate entry '0' for key 'PRIMARY'
[Method: POST, Route: agregarAutores]
in SYSTEMPATH\Database\BaseConnection.php on line 684.
 1 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 2 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 3 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 4 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 5 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 6 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
 7 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
 8 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
11 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 15:43:55 --> [Caused by] CodeIgniter\Database\Exceptions\DatabaseException: Duplicate entry '0' for key 'PRIMARY'
in SYSTEMPATH\Database\MySQLi\Connection.php on line 332.
 1 SYSTEMPATH\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'0\', \'vale\', \'aa\', \'aa\')')
 2 SYSTEMPATH\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'0\', \'vale\', \'aa\', \'aa\')')
 3 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 4 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 5 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 6 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 7 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 8 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
 9 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
10 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
11 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
12 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
13 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 15:43:55 --> [Caused by] mysqli_sql_exception: Duplicate entry '0' for key 'PRIMARY'
in SYSTEMPATH\Database\MySQLi\Connection.php on line 327.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(327): mysqli->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'0\', \'vale\', \'aa\', \'aa\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'0\', \'vale\', \'aa\', \'aa\')')
 3 SYSTEMPATH\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'0\', \'vale\', \'aa\', \'aa\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 5 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 6 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 7 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 8 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 9 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
10 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
11 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
13 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
14 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 15:55:45 --> ErrorException: Undefined array key "codigo_autor"
[Method: GET, Route: autores]
in APPPATH\Views\vista_autores.php on line 35.
 1 APPPATH\Views\vista_autores.php(35): CodeIgniter\Debug\Exceptions->errorHandler(2, 'Undefined array key "codigo_autor"', 'C:\\xampp\\htdocs\\fs2025\\biblioteca\\codeigniter4-framework-68d1a58\\app\\Views\\vista_autores.php', 35)
 2 SYSTEMPATH\View\View.php(226): include('C:\\xampp\\htdocs\\fs2025\\biblioteca\\codeigniter4-framework-68d1a58\\app\\Views\\vista_autores.php')
 3 SYSTEMPATH\View\View.php(229): CodeIgniter\View\View->CodeIgniter\View\{closure}()
 4 SYSTEMPATH\Common.php(1167): CodeIgniter\View\View->render('vista_autores', [], true)
 5 APPPATH\Controllers\Autores.php(14): view('vista_autores', [...])
 6 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->vistaAutores()
 7 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
 8 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
11 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
ERROR - 2025-08-25 16:52:59 --> mysqli_sql_exception: Duplicate entry '123' for key 'PRIMARY' in C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\MySQLi\Connection.php:327
Stack trace:
#0 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\MySQLi\Connection.php(327): mysqli->query('INSERT INTO `au...', 0)
#1 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `au...')
#2 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `au...')
#3 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `au...', Array, false)
#4 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
#5 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\BaseModel.php(838): CodeIgniter\Model->doInsert(Array)
#6 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Model.php(797): CodeIgniter\BaseModel->insert(Array, true)
#7 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\app\Controllers\Autores.php(28): CodeIgniter\Model->insert(Array)
#8 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
#9 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
#10 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Boot.php(363): CodeIgniter\CodeIgniter->run()
#12 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
#13 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\public\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
#14 {main}
CRITICAL - 2025-08-25 16:52:59 --> CodeIgniter\Database\Exceptions\DatabaseException: Duplicate entry '123' for key 'PRIMARY'
[Method: POST, Route: agregarAutores]
in SYSTEMPATH\Database\BaseConnection.php on line 684.
 1 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 2 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 3 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 4 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 5 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 6 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
 7 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
 8 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
11 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 16:52:59 --> [Caused by] CodeIgniter\Database\Exceptions\DatabaseException: Duplicate entry '123' for key 'PRIMARY'
in SYSTEMPATH\Database\MySQLi\Connection.php on line 332.
 1 SYSTEMPATH\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'123\', \'ruiz\', \'Beatriz\', \'salvador\')')
 2 SYSTEMPATH\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'123\', \'ruiz\', \'Beatriz\', \'salvador\')')
 3 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 4 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 5 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 6 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 7 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 8 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
 9 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
10 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
11 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
12 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
13 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 16:52:59 --> [Caused by] mysqli_sql_exception: Duplicate entry '123' for key 'PRIMARY'
in SYSTEMPATH\Database\MySQLi\Connection.php on line 327.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(327): mysqli->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'123\', \'ruiz\', \'Beatriz\', \'salvador\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'123\', \'ruiz\', \'Beatriz\', \'salvador\')')
 3 SYSTEMPATH\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'123\', \'ruiz\', \'Beatriz\', \'salvador\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 5 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 6 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 7 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 8 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 9 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
10 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
11 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
13 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
14 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
ERROR - 2025-08-25 16:53:10 --> mysqli_sql_exception: Duplicate entry '123' for key 'PRIMARY' in C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\MySQLi\Connection.php:327
Stack trace:
#0 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\MySQLi\Connection.php(327): mysqli->query('INSERT INTO `au...', 0)
#1 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `au...')
#2 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `au...')
#3 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `au...', Array, false)
#4 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
#5 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\BaseModel.php(838): CodeIgniter\Model->doInsert(Array)
#6 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Model.php(797): CodeIgniter\BaseModel->insert(Array, true)
#7 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\app\Controllers\Autores.php(28): CodeIgniter\Model->insert(Array)
#8 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
#9 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
#10 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Boot.php(363): CodeIgniter\CodeIgniter->run()
#12 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
#13 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\public\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
#14 {main}
CRITICAL - 2025-08-25 16:53:10 --> CodeIgniter\Database\Exceptions\DatabaseException: Duplicate entry '123' for key 'PRIMARY'
[Method: POST, Route: agregarAutores]
in SYSTEMPATH\Database\BaseConnection.php on line 684.
 1 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 2 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 3 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 4 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 5 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 6 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
 7 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
 8 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
11 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 16:53:10 --> [Caused by] CodeIgniter\Database\Exceptions\DatabaseException: Duplicate entry '123' for key 'PRIMARY'
in SYSTEMPATH\Database\MySQLi\Connection.php on line 332.
 1 SYSTEMPATH\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'123\', \'ruiz\', \'Beatriz\', \'salvador\')')
 2 SYSTEMPATH\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'123\', \'ruiz\', \'Beatriz\', \'salvador\')')
 3 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 4 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 5 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 6 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 7 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 8 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
 9 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
10 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
11 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
12 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
13 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 16:53:10 --> [Caused by] mysqli_sql_exception: Duplicate entry '123' for key 'PRIMARY'
in SYSTEMPATH\Database\MySQLi\Connection.php on line 327.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(327): mysqli->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'123\', \'ruiz\', \'Beatriz\', \'salvador\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'123\', \'ruiz\', \'Beatriz\', \'salvador\')')
 3 SYSTEMPATH\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'123\', \'ruiz\', \'Beatriz\', \'salvador\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 5 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 6 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 7 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 8 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 9 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutores()
10 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
11 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
13 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
14 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 17:12:26 --> ParseError: syntax error, unexpected end of file
[Method: GET, Route: autores]
in APPPATH\Config\Routes.php on line 18.
 1 SYSTEMPATH\CodeIgniter.php(821): CodeIgniter\Router\RouteCollection->loadRoutes()
 2 SYSTEMPATH\CodeIgniter.php(455): CodeIgniter\CodeIgniter->tryToRouteIt(null)
 3 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 4 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
 5 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
 6 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 17:12:43 --> ParseError: syntax error, unexpected end of file
[Method: GET, Route: autores]
in APPPATH\Config\Routes.php on line 18.
 1 SYSTEMPATH\CodeIgniter.php(821): CodeIgniter\Router\RouteCollection->loadRoutes()
 2 SYSTEMPATH\CodeIgniter.php(455): CodeIgniter\CodeIgniter->tryToRouteIt(null)
 3 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 4 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
 5 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
 6 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 17:12:50 --> ParseError: syntax error, unexpected end of file
[Method: GET, Route: autores]
in APPPATH\Config\Routes.php on line 18.
 1 SYSTEMPATH\CodeIgniter.php(821): CodeIgniter\Router\RouteCollection->loadRoutes()
 2 SYSTEMPATH\CodeIgniter.php(455): CodeIgniter\CodeIgniter->tryToRouteIt(null)
 3 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 4 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
 5 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
 6 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 17:13:24 --> ParseError: syntax error, unexpected end of file
[Method: GET, Route: /]
in APPPATH\Config\Routes.php on line 18.
 1 SYSTEMPATH\CodeIgniter.php(821): CodeIgniter\Router\RouteCollection->loadRoutes()
 2 SYSTEMPATH\CodeIgniter.php(455): CodeIgniter\CodeIgniter->tryToRouteIt(null)
 3 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 4 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
 5 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
 6 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 17:13:29 --> ParseError: syntax error, unexpected end of file
[Method: GET, Route: Home]
in APPPATH\Config\Routes.php on line 18.
 1 SYSTEMPATH\CodeIgniter.php(821): CodeIgniter\Router\RouteCollection->loadRoutes()
 2 SYSTEMPATH\CodeIgniter.php(455): CodeIgniter\CodeIgniter->tryToRouteIt(null)
 3 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 4 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
 5 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
 6 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 17:14:03 --> Error: Call to undefined function get()
[Method: GET, Route: autores]
in APPPATH\Config\Routes.php on line 10.
 1 SYSTEMPATH\Router\RouteCollection.php(339): require()
 2 SYSTEMPATH\CodeIgniter.php(821): CodeIgniter\Router\RouteCollection->loadRoutes()
 3 SYSTEMPATH\CodeIgniter.php(455): CodeIgniter\CodeIgniter->tryToRouteIt(null)
 4 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 5 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
 6 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
 7 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 17:14:21 --> Error: Call to undefined function get()
[Method: GET, Route: autores]
in APPPATH\Config\Routes.php on line 10.
 1 SYSTEMPATH\Router\RouteCollection.php(339): require()
 2 SYSTEMPATH\CodeIgniter.php(821): CodeIgniter\Router\RouteCollection->loadRoutes()
 3 SYSTEMPATH\CodeIgniter.php(455): CodeIgniter\CodeIgniter->tryToRouteIt(null)
 4 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 5 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
 6 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
 7 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 17:14:21 --> Error: Call to undefined function get()
[Method: GET, Route: autores]
in APPPATH\Config\Routes.php on line 10.
 1 SYSTEMPATH\Router\RouteCollection.php(339): require()
 2 SYSTEMPATH\CodeIgniter.php(821): CodeIgniter\Router\RouteCollection->loadRoutes()
 3 SYSTEMPATH\CodeIgniter.php(455): CodeIgniter\CodeIgniter->tryToRouteIt(null)
 4 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 5 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
 6 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
 7 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 17:14:21 --> Error: Call to undefined function get()
[Method: GET, Route: autores]
in APPPATH\Config\Routes.php on line 10.
 1 SYSTEMPATH\Router\RouteCollection.php(339): require()
 2 SYSTEMPATH\CodeIgniter.php(821): CodeIgniter\Router\RouteCollection->loadRoutes()
 3 SYSTEMPATH\CodeIgniter.php(455): CodeIgniter\CodeIgniter->tryToRouteIt(null)
 4 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 5 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
 6 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
 7 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 17:14:22 --> Error: Call to undefined function get()
[Method: GET, Route: autores]
in APPPATH\Config\Routes.php on line 10.
 1 SYSTEMPATH\Router\RouteCollection.php(339): require()
 2 SYSTEMPATH\CodeIgniter.php(821): CodeIgniter\Router\RouteCollection->loadRoutes()
 3 SYSTEMPATH\CodeIgniter.php(455): CodeIgniter\CodeIgniter->tryToRouteIt(null)
 4 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 5 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
 6 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
 7 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 17:14:22 --> Error: Call to undefined function get()
[Method: GET, Route: autores]
in APPPATH\Config\Routes.php on line 10.
 1 SYSTEMPATH\Router\RouteCollection.php(339): require()
 2 SYSTEMPATH\CodeIgniter.php(821): CodeIgniter\Router\RouteCollection->loadRoutes()
 3 SYSTEMPATH\CodeIgniter.php(455): CodeIgniter\CodeIgniter->tryToRouteIt(null)
 4 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 5 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
 6 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
 7 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 17:14:23 --> Error: Call to undefined function get()
[Method: GET, Route: autores]
in APPPATH\Config\Routes.php on line 10.
 1 SYSTEMPATH\Router\RouteCollection.php(339): require()
 2 SYSTEMPATH\CodeIgniter.php(821): CodeIgniter\Router\RouteCollection->loadRoutes()
 3 SYSTEMPATH\CodeIgniter.php(455): CodeIgniter\CodeIgniter->tryToRouteIt(null)
 4 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 5 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
 6 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
 7 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 17:14:23 --> Error: Call to undefined function get()
[Method: GET, Route: autores]
in APPPATH\Config\Routes.php on line 10.
 1 SYSTEMPATH\Router\RouteCollection.php(339): require()
 2 SYSTEMPATH\CodeIgniter.php(821): CodeIgniter\Router\RouteCollection->loadRoutes()
 3 SYSTEMPATH\CodeIgniter.php(455): CodeIgniter\CodeIgniter->tryToRouteIt(null)
 4 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 5 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
 6 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
 7 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
ERROR - 2025-08-25 17:39:21 --> mysqli_sql_exception: Duplicate entry '128' for key 'PRIMARY' in C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\MySQLi\Connection.php:327
Stack trace:
#0 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\MySQLi\Connection.php(327): mysqli->query('INSERT INTO `au...', 0)
#1 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `au...')
#2 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `au...')
#3 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `au...', Array, false)
#4 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
#5 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\BaseModel.php(838): CodeIgniter\Model->doInsert(Array)
#6 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Model.php(797): CodeIgniter\BaseModel->insert(Array, true)
#7 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\app\Controllers\Autores.php(28): CodeIgniter\Model->insert(Array)
#8 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(933): App\Controllers\Autores->agregarAutor()
#9 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
#10 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Boot.php(363): CodeIgniter\CodeIgniter->run()
#12 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
#13 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\public\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
#14 {main}
CRITICAL - 2025-08-25 17:39:21 --> CodeIgniter\Database\Exceptions\DatabaseException: Duplicate entry '128' for key 'PRIMARY'
[Method: POST, Route: agregarAutor]
in SYSTEMPATH\Database\BaseConnection.php on line 684.
 1 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 2 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 3 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 4 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 5 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 6 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutor()
 7 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
 8 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
11 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 17:39:21 --> [Caused by] CodeIgniter\Database\Exceptions\DatabaseException: Duplicate entry '128' for key 'PRIMARY'
in SYSTEMPATH\Database\MySQLi\Connection.php on line 332.
 1 SYSTEMPATH\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'128\', \'Gonzalez\', \'aa\', \'honduras\')')
 2 SYSTEMPATH\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'128\', \'Gonzalez\', \'aa\', \'honduras\')')
 3 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 4 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 5 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 6 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 7 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 8 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutor()
 9 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
10 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
11 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
12 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
13 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 17:39:21 --> [Caused by] mysqli_sql_exception: Duplicate entry '128' for key 'PRIMARY'
in SYSTEMPATH\Database\MySQLi\Connection.php on line 327.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(327): mysqli->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'128\', \'Gonzalez\', \'aa\', \'honduras\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'128\', \'Gonzalez\', \'aa\', \'honduras\')')
 3 SYSTEMPATH\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'128\', \'Gonzalez\', \'aa\', \'honduras\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 5 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 6 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 7 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 8 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 9 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutor()
10 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
11 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
13 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
14 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
ERROR - 2025-08-25 17:39:58 --> mysqli_sql_exception: Duplicate entry '128' for key 'PRIMARY' in C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\MySQLi\Connection.php:327
Stack trace:
#0 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\MySQLi\Connection.php(327): mysqli->query('INSERT INTO `au...', 0)
#1 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `au...')
#2 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `au...')
#3 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `au...', Array, false)
#4 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
#5 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\BaseModel.php(838): CodeIgniter\Model->doInsert(Array)
#6 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Model.php(797): CodeIgniter\BaseModel->insert(Array, true)
#7 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\app\Controllers\Autores.php(28): CodeIgniter\Model->insert(Array)
#8 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(933): App\Controllers\Autores->agregarAutor()
#9 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
#10 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Boot.php(363): CodeIgniter\CodeIgniter->run()
#12 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\system\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
#13 C:\xampp\htdocs\fs2025\biblioteca\codeigniter4-framework-68d1a58\public\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
#14 {main}
CRITICAL - 2025-08-25 17:39:58 --> CodeIgniter\Database\Exceptions\DatabaseException: Duplicate entry '128' for key 'PRIMARY'
[Method: POST, Route: agregarAutor]
in SYSTEMPATH\Database\BaseConnection.php on line 684.
 1 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 2 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 3 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 4 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 5 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 6 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutor()
 7 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
 8 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
11 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 17:39:58 --> [Caused by] CodeIgniter\Database\Exceptions\DatabaseException: Duplicate entry '128' for key 'PRIMARY'
in SYSTEMPATH\Database\MySQLi\Connection.php on line 332.
 1 SYSTEMPATH\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'128\', \'Gonzalez\', \'aa\', \'honduras\')')
 2 SYSTEMPATH\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'128\', \'Gonzalez\', \'aa\', \'honduras\')')
 3 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 4 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 5 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 6 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 7 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 8 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutor()
 9 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
10 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
11 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
12 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
13 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
CRITICAL - 2025-08-25 17:39:58 --> [Caused by] mysqli_sql_exception: Duplicate entry '128' for key 'PRIMARY'
in SYSTEMPATH\Database\MySQLi\Connection.php on line 327.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(327): mysqli->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'128\', \'Gonzalez\', \'aa\', \'honduras\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(729): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'128\', \'Gonzalez\', \'aa\', \'honduras\')')
 3 SYSTEMPATH\Database\BaseConnection.php(646): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (\'128\', \'Gonzalez\', \'aa\', \'honduras\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(2345): CodeIgniter\Database\BaseConnection->query('INSERT INTO `autores` (`codigo_autor`, `apellido`, `nombre`, `nacionalidad`) VALUES (:codigo_autor:, :apellido:, :nombre:, :nacionalidad:)', [...], false)
 5 SYSTEMPATH\Model.php(394): CodeIgniter\Database\BaseBuilder->insert()
 6 SYSTEMPATH\BaseModel.php(838): CodeIgniter\Model->doInsert([...])
 7 SYSTEMPATH\Model.php(797): CodeIgniter\BaseModel->insert([...], true)
 8 APPPATH\Controllers\Autores.php(28): CodeIgniter\Model->insert([...])
 9 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\Autores->agregarAutor()
10 SYSTEMPATH\CodeIgniter.php(507): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Autores))
11 SYSTEMPATH\CodeIgniter.php(354): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
12 SYSTEMPATH\Boot.php(363): CodeIgniter\CodeIgniter->run()
13 SYSTEMPATH\Boot.php(68): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
14 FCPATH\index.php(59): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
